generate/genqlient_directive.go

```diff
@@ -244,68 +361,116 @@ func validateFlattenOption(
 	return index, nil
 }
 
-// merge joins the directive applied to this node (the argument) and the one
-// applied to the entire operation (the receiver) and returns a new
-// directive-object representing the options to apply to this node (where in
-// general we take the node's option, then the operation's, then the default).
-func (dir *genqlientDirective) merge(other *genqlientDirective) *genqlientDirective {
-	retval := *dir
-	if other.Omitempty != nil {
-		retval.Omitempty = other.Omitempty
+func fillDefaultBool(target **bool, defaults ...*bool) {
+	if *target != nil {
+		return
 	}
-	if other.Pointer != nil {
-		retval.Pointer = other.Pointer
+
+	for _, val := range defaults {
+		if val != nil {
+			*target = val
+			return
+		}
 	}
-	if other.Struct != nil {
-		retval.Struct = other.Struct
+}
+
+func fillDefaultString(target *string, defaults ...string) {
+	if *target != "" {
+		return
+	}
+
+	for _, val := range defaults {
+		if val != "" {
+			*target = val
+			return
+		}
 	}
-	if other.Flatten != nil {
-		retval.Flatten = other.Flatten
+}
+
+// merge updates the receiver, which is a directive applied to some node, with
+// the information from the directive applied to the fragment or operation
+// containing that node.  (The update is in-place.)
+//
+// Note this has slightly different semantics than .add(), see inline for
+// details.
+//
+// parent is as described in parsePrecedingComment.
+func (dir *genqlientDirective) mergeOperationDirective(
+	node interface{},
+	parent *ast.Definition,
+	operationDirective *genqlientDirective,
+) {
+	var forField *genqlientDirective
```

Comment:
This is the `for` field for some parent element whose `for` points to us, is that right?

Reply:
Yes, I'll comment better.